api/src/pcapi/core/offers/models.py

```diff
@@ -322,14 +329,14 @@ class WithdrawalTypeEnum(enum.Enum):
     ON_SITE = "on_site"
 
 
-class Offer(PcObject, Model, ExtraDataMixin, DeactivableMixin, ValidationMixin, AccessibilityMixin, StatusMixin):  # type: ignore [valid-type, misc]
+class Offer(PcObject, Base, Model, ExtraDataMixin, DeactivableMixin, ValidationMixin, AccessibilityMixin, StatusMixin):  # type: ignore [valid-type, misc]
     __tablename__ = "offer"
 
     id = sa.Column(sa.BigInteger, primary_key=True, autoincrement=True)
 
     productId = sa.Column(sa.BigInteger, sa.ForeignKey("product.id"), index=True, nullable=False)
 
-    product = sa.orm.relationship("Product", foreign_keys=[productId], backref="offers")
+    product: Product | None = sa.orm.relationship(Product, foreign_keys=[productId], backref="offers")
```

Comment:
Le champ n'est pas nullable, donc je pense que le typage devrait plutôt être : 

```suggestion
    product: Product = sa.orm.relationship(Product, foreign_keys=[productId], backref="offers")
```